src/main/java/com/alipay/oceanbase/hbase/OHTable.java

```diff
@@ -1691,68 +1835,68 @@ private KeyValue modifyQualifier(Cell original, byte[] newQualifier) {
         long timestamp = original.getTimestamp();
         KeyValue.Type type = KeyValue.Type.codeToType(original.getType().getCode());
         // Create a new KeyValue with the modified qualifier  
-        return new KeyValue(row, family, newQualifier, timestamp, type,
-            value);
+        return new KeyValue(row, family, newQualifier, timestamp, type, value);
     }
 
-    private BatchOperation buildBatchOperation(String tableName, List<? extends Row> actions, boolean isTableGroup, List<Integer> resultMapSingleOp) {
+    private BatchOperation buildBatchOperation(String tableName, List<? extends Row> actions,
+                                               boolean isTableGroup, List<Integer> resultMapSingleOp)
+                                                                                                     throws FeatureNotSupportedException,
+                                                                                                     IllegalArgumentException {
         BatchOperation batch = obTableClient.batchOperation(tableName);
-        if (actions != null) {
-            int posInList = -1;
-            int singleOpResultNum;
-            for (Row row : actions) {
-                singleOpResultNum = 0;
-                posInList++;
-                if (!(row instanceof Put) && !(row instanceof Delete)) {
-                    throw new FeatureNotSupportedException(
-                        "not supported other type in batch yet,only support put and delete");
-                } else if (row instanceof Put) {
-                    Put put = (Put) row;
-                    if (put.isEmpty()) {
-                        throw new IllegalArgumentException("No columns to insert for #"
-                                                           + (posInList + 1) + " item");
+        int posInList = -1;
+        int singleOpResultNum;
+        for (Row row : actions) {
+            singleOpResultNum = 0;
+            posInList++;
+            if (row instanceof Put) {
+                Put put = (Put) row;
+                if (put.isEmpty()) {
+                    throw new IllegalArgumentException("No columns to insert for #"
+                            + (posInList + 1) + " item");
+                }
+                for (Map.Entry<byte[], List<Cell>> entry : put.getFamilyCellMap().entrySet()) {
+                    byte[] family = entry.getKey();
+                    List<Cell> keyValueList = entry.getValue();
+                    for (Cell kv : keyValueList) {
+                        singleOpResultNum++;
+                        if(isTableGroup){
```

Comment:
can we handle tablegroup requset in buildMutation like Delete ?

Reply:
fixed